### ISSUE TITLE
bitfield access-generated code is disassembled with incorrect addends

### ISSUE THREAD

**Issue by EllipticEllipsis** [Decompollaborate/spimdisasm#110]
Minimal example:
```c
struct {
    u32 f0: 8;
    u32 f1: 8;
} test;

void set_f1(u32 x) {
    test.f1 = x;
}
```
compile with IDO (5.3 or 7.1), `cc bitfield_test.c -c -non_shared -G 0 -O1`

`-S` output:
```mips
	.globl	set_f2
	.loc	2 70
 #  70	void set_f2(u32 x) {
	.ent	set_f2 2
set_f2:
	.option	O1
	.frame	$sp, 0, $31
	.loc	2 70
	.loc	2 71
 #  71	    test.f2 = x;
	sb	$4, test+1
	.loc	2 72
 #  72	}
	.livereg	0x0000FF0E,0x00000FFF
	j	$31
	.end	set_f2

```

Objdump output:
```
000002f8 <set_f2>:
 2f8:	3c010000 	lui	at,0x0
			2f8: R_MIPS_HI16	test
 2fc:	03e00008 	jr	ra
 300:	a0240001 	sb	a0,1(at)
			300: R_MIPS_LO16	test
 304:	00000000 	nop
```

Current elfObjDisasm output:
```mips
glabel set_f2 # 22
  lui         $at, %hi(test)
  jr          $ra
   sb         $a0, %lo(test)($at)
  nop
```
Should probably be
```mips
glabel set_f2 # 22
  lui         $at, %hi(test+1)
  jr          $ra
   sb         $a0, %lo(test+1)($at)
  nop
```

**Comment by AngheloAlf** [comment]
Just pushed a fix to the develop branch, could you check if it properly fixes all your use cases?

It seems like spimdisasm wasn't using the addend on non `static` symbols, whoops

**Comment by EllipticEllipsis** [comment]
I think so, for example
```
glabel set_bitfield_2_0180 # 26
  lui         $t9, %hi(b2_1 + 0x2)
  lhu         $t9, %lo(b2_1 + 0x2)($t9)
  andi        $t6, $a0, 0x3
  sll         $t7, $t6, 7
  andi        $t8, $t7, 0x180
  andi        $t0, $t9, 0xFE7F
  or          $t1, $t8, $t0
  lui         $at, %hi(b2_1 + 0x2)
  jr          $ra
   sh         $t1, %lo(b2_1 + 0x2)($at)
```
now uses the addend correctly.

**Comment by AngheloAlf** [comment]
Fixed in release 1.12.2
